presto-hive/src/main/java/io/prestosql/plugin/hive/util/SerDeUtils.java

```diff
@@ -114,25 +104,25 @@ private static void serializePrimitive(Type type, BlockBuilder builder, Object o
 
         switch (inspector.getPrimitiveCategory()) {
             case BOOLEAN:
-                BooleanType.BOOLEAN.writeBoolean(builder, ((BooleanObjectInspector) inspector).get(object));
+                type.writeBoolean(builder, ((BooleanObjectInspector) inspector).get(object));
```

Comment:
"Use the passed type to serialize in SerDeUtils" commit looks good. Can you split it out to a separate PR to reduce scope of this PR?

Reply:
[Done.](https://github.com/prestosql/presto/pull/6175)